lib/server.go

```diff
@@ -1400,6 +1415,27 @@ func (srv *Server) _handleSnapshot(pp *Peer, msg *MsgDeSoSnapshotData) {
 	srv.GetBlocks(pp, int(headerTip.Height))
 }
 
+// updateDbOpts closes the current badger DB instance and re-opens it with the provided new options.
+func (srv *Server) updateDbOpts(opts badger.Options) {
+	// Make sure that a mempool process doesn't try to access the DB while we're closing and re-opening it.
+	srv.mempool.mtx.RLock()
+	defer srv.mempool.mtx.RUnlock()
```

Comment:
You may actually want Lock() here not RLock(). RLock() is non-exclusive meaning that other threads could be reading the mempool db (but not writing to it). If you do Lock() it kicks out readers in addition to writers.